### PR TITLE
Bugs/parquet created at 179252091

### PR DESCRIPTION
**Pull request by rosalinekarr**
Problem
=======
Our parquet schema does not match the spec
[#179252091](https://www.pivotaltracker.com/story/show/179252091)

Solution
========
Update the schema to match the spec

Change summary:
---------------
* Added `createdAt` to the parquet schema for each announcement type
* Added `contentHash` to the parquet schema for profile announcements
* Added Tombstones to the parquet schema with tests
* Updated types on parquet schema to require all announcement types be defined

### CODE REVIEW

File: src/core/batch/parquetSchema.ts [hunk 1]
```diff
@@ -26,6 +26,25 @@ type columnName = string;
...
+  targetSignature: { type: "BYTE_ARRAY" },
+  signature: { type: "BYTE_ARRAY" },
+  createdAt: { type: "INT64" },
+};
+
+/**
+ * TombstoneBloomFilter: bloom filter options for batching broadcast announcements
+ */
+export const TombstoneBloomFilterOptions: BloomFilterOptions = {
+  bloomFilters: [{ column: "fromId" }],
```
Comment (wilwade) [inline src/core/batch/parquetSchema.ts:45]: Also `targetSignature`

File: src/core/batch/parquetSchema.ts [hunk 2]
```diff
@@ -127,20 +151,18 @@ export const ReactionBloomFilterOptions = {
...
+    [AnnouncementType.GraphChange]: GraphChangeSchema,
+    [AnnouncementType.Broadcast]: BroadcastSchema,
+    [AnnouncementType.Reply]: ReplySchema,
+    [AnnouncementType.Reaction]: ReactionSchema,
+    [AnnouncementType.Profile]: ProfileSchema,
+  };
+
+  if (schemas[announcementType] === undefined) throw new InvalidAnnouncementTypeError(announcementType);
+
+  return schemas[announcementType];
```
Comment (wilwade) [inline src/core/batch/parquetSchema.ts:165]: Shorter, but harder to read perhaps
```suggestion
  return schemas[announcementType] || throw new InvalidAnnouncementTypeError(announcementType);
```

File: src/core/batch/parquetSchema.ts (hunk 1 above)
Comment (rosalinekarr) [reply src/core/batch/parquetSchema.ts]: fixed

File: src/core/batch/parquetSchema.ts [hunk 3]
```diff
@@ -127,20 +151,18 @@ export const ReactionBloomFilterOptions = {
...
-    case AnnouncementType.Reply:
-      return ReplySchema;
-    case AnnouncementType.Reaction:
-      return ReactionSchema;
-    case AnnouncementType.Profile:
-      return ProfileSchema;
-  }
-
-  throw new InvalidAnnouncementTypeError(announcementType);
+  const schemas: Record<AnnouncementType, Schema> = {
```
Comment (wilwade) [inline src/core/batch/parquetSchema.ts:154]: 🎉

File: src/core/batch/parquetSchema.ts (hunk 2 above)
Comment (rosalinekarr) [reply src/core/batch/parquetSchema.ts:165]: The parser doesn't seem to like it:

```
/Users/rosalinekarr/Development/Liberty/sdk-ts/src/core/batch/parquetSchema.ts
  163:38  error  Parsing error: Expression expected
```

**Review by enddynayn — APPROVED** [review]

**Pull request merged**